src/js/profile/wearable/widget/wearable/TimePicker.js

```diff
@@ -472,7 +472,7 @@
 				var self = this,
 					parentElement = getClosestByClass(event.target, classes.CONTAINER);
 
-				if (parentElement) {
+				if (event.target.classList.contains(classes.ACTIVE_LABEL) && parentElement) {
```

Comment:
I cannot reproduce the issue but I understand that the issue occurred.
I have suggestion to change:
```
				var self = this,
					spin = null,
					parentElement = getClosestByClass(event.target, classes.CONTAINER);

				spin = ns.engine.getBinding(event.target);
				if (spin && spin.option("enabled") && parentElement) {
					self.value(event.detail.value);
				}
```

Reply:
I verified this PR on device... but unfortunately seems like doesn't work.

Reply:
@TomaszLukawskiSam ,
Yes.. It seem to only work on the TimePicker correctly.